File: go/sort/bubble_sort.go
package main

import (
	"fmt"
	"go/util"
)

func main() {

	arr := []int{1, 43, 132, 5, 89, 122, 546, 8, 9}

	bubble_sort(arr)

	fmt.Println(arr)
}

func bubble_sort(arr []int) {

	var i int
	for ; i < len(arr); i++ {

		var j = i + 1
		var boundary = len(arr) - 1

		for ; j <= boundary; j++ {

			if arr[i] > arr[j] {
				util.Swap(arr, i, j)
			}
		}

		boundary--

	}
}

File: py/sort/bubble_sort.py
from sort import *


def bubble_sort(arr):
    i = 0
    length = len(arr)
    while i < length:

        j = i + 1
        boundary = length - 1

        while j <= boundary:

            if arr[i] > arr[j]:
                swap(arr, i, j)

            j = self_incr(j)

        i = self_incr(i)


if __name__ == '__main__':
    arr = [1, 3, 5, 13, 132, 789, 12, 4, 9]

    bubble_sort(arr)

    print(arr)
